### DISCUSSION TITLE
Fallback image is always selected in inspector tools

### DISCUSSION THREAD

**Discussion by notflip** [spatie/statamic-responsive-images#182] [Q&A] [answered]
I'm not sure if the package is incorrectly setup, or if this is normal chrome inspector behaviour, but the fallback image is always selected, at any size.

![Screenshot 2022-11-08 at 10 46 36](https://user-images.githubusercontent.com/4146384/200533173-e164d898-fdc4-4148-b74f-421711172e7b.png)

Does anyone know what is up here?

**Comment by ncla** [comment, marked as answer]
It is probably because your `media` attribute is empty for each `<source>`, it should have the breakpoint queries (something like `min-width: 1280px` for example).

**Reply by notflip** [reply]
Thanks @ncla Is this something I have to configure?

**Reply by ncla** [reply]
It should work by default, especially if you have the addons config published in your project. These are the default breakpoints. https://github.com/spatie/statamic-responsive-images/blob/main/config/responsive-images.php#L102-L108

If it doesn't work with those default breakpoints then this might be a bug.

**Reply by notflip** [reply]
For the following setup (screenshot)
![Screenshot 2022-11-08 at 12 25 16](https://user-images.githubusercontent.com/4146384/200552187-7b677475-b185-418d-9cdf-feed1def1a49.png)

Which outputs 6 sources, of which 4 have the media attribute set, 2 have an empty media  in order of appearance

- media="(min-width: 1024px)"
- media="(min-width: 1024px)"
- media="(min-width: 640px)"
- media="(min-width: 640px)"
- blank
- blank

**Reply by ncla** [reply]
Ah right, I think media attribute being empty is normal if you use default breakpoint only.

What you described then sounds like it is not an issue with `media` attribute at all. Two `<source>` tags for WEBP and AVIF format, three breakpoints in total. This looks correct to me.

How do you verify that the correct image was loaded by browser?

When you say "fallback image", do you mean the placeholder blurry image or the one defined at `src` attribute on `<img>` tag?

**Reply by notflip** [reply]
I mean the image tag src image, thanks!

**Reply by ncla** [reply]
How do you verify what image was loaded by browser? You should look at devtool network requests and check URLs.

**Reply by notflip** [reply]
I'll check it out thanks, might be the inspector that is not highlighting the correct source tag that's being used

**Reply by ncla** [reply]
Slow day for me.. but I think I understood you now finally. If you right click on web page and inspect element, then yes, the focused element being `<img>` is normal. The focused element does not equate to which source or image was loaded.

The reason it does not focus on `<picture>` may be to do with how it behaves:

> the [picture](https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element) element itself does not display anything; it merely provides a context for its contained [img](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element) element that enables it to choose from multiple [URLs](https://url.spec.whatwg.org/#concept-url).

https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element

**Reply by notflip** [reply]
Thanks @ncla, you're absolutely right, in the network tab, and while testing with crop modes I can see the image changing, lovely! This package is amazing. Thank you.
